internal/cmd/kafka/command_cluster_create.go

```diff
@@ -142,32 +147,52 @@ func (c *clusterCommand) create(cmd *cobra.Command, args []string, prompt form.P
 		return err
 	}
 
-	encryptionKey, err := cmd.Flags().GetString("encryption-key")
+	var encryptionKey *string
+	if cmd.Flags().Changed("encryption-key") {
+		*encryptionKey, err = cmd.Flags().GetString("encryption-key")
+		if err != nil {
+			return err
+		}
+
+		if *encryptionKey != "" {
+			input := validateEncryptionKeyInput{
+				Cloud:          cloud,
+				MetadataClouds: clouds,
+				AccountID:      c.EnvironmentId(),
+			}
+			if err := c.validateEncryptionKey(cmd, prompt, input); err != nil {
+				return err
+			}
+		}
+	}
+
+	byok, err := cmd.Flags().GetString("byok")
 	if err != nil {
 		return err
 	}
 
-	if encryptionKey != "" {
-		input := validateEncryptionKeyInput{
-			Cloud:          cloud,
-			MetadataClouds: clouds,
-			AccountID:      c.EnvironmentId(),
+	var keyGlobalObjectReference *cmkv2.GlobalObjectReference
+	if byok != "" {
+		key, httpResp, err := c.V2Client.GetByokKey(byok)
+		if err != nil {
+			return errors.CatchByokKeyNotFoundError(err, httpResp)
 		}
-		if err := c.validateEncryptionKey(cmd, prompt, input); err != nil {
-			return err
+		keyGlobalObjectReference = &cmkv2.GlobalObjectReference{
+			Id: *key.Id,
 		}
 	}
 
 	createCluster := cmkv2.CmkV2Cluster{
 		Spec: &cmkv2.CmkV2ClusterSpec{
-			Environment: &cmkv2.ObjectReference{
+			Environment: &cmkv2.EnvScopedObjectReference{
 				Id: c.EnvironmentId(),
 			},
```

Comment:
Would be great if you could squeeze this onto one line, since there's only one field.
```suggestion
			Environment: &cmkv2.EnvScopedObjectReference{Id: c.EnvironmentId()},
```

Reply:
Fixed.